onnxruntime/core/providers/cuda/nn/conv.cc

```diff
@@ -176,7 +180,7 @@ Status Conv<T, NHWC>::UpdateState(OpKernelContext* context, bool bias_expected)
 
     if (w_dims_changed) {
       s_.last_w_dims = gsl::make_span(w_dims);
-      s_.cached_benchmark_results.clear();
+      if constexpr (!NHWC) s_.cached_benchmark_results.clear();
```

Comment:
From my understanding, clear benchmark result does not depend on whether it is NHWC or not. If the weight dimension changes, old benchmark results shall not apply on new weights.

In reality, weights is usually initializer and weight shape does not change, and this code branch will not be called unless someone creates a model manually to use dynamic shape weights to test Conv.

Reply:
As NHWC is a template parameter this is not subject to change for the same class instance.  Benchmark results can change significantly depending on the data layout though.

Reply:
I thought there is no need to change this line unless NHWC has different way to cache benchmark results.

Reply:
On NHWC we do not store the benchmark results but a cudnn frontend graph, which contains a compiled execution plan. On a input or weight dimension change we build a new execution plan.